packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -1235,10 +1235,17 @@ class _SearchBarState extends State<SearchBar> {
                         controller: widget.controller,
                         style: effectiveTextStyle,
                         decoration: InputDecoration(
-                          border: InputBorder.none,
                           hintText: widget.hintText,
+                        ).applyDefaults(InputDecorationTheme(
                           hintStyle: effectiveHintStyle,
-                        ),
+
+                          // The configuration below is to make sure that the text field
+                          // in `SearchBar` will not be overridden by the overall `InputDecorationTheme`
+                          enabledBorder: InputBorder.none,
+                          border: InputBorder.none,
+                          focusedBorder: InputBorder.none,
+                          contentPadding: const EdgeInsets.fromLTRB(0.0, 12.0, 0.0, 12.0),
```

Comment:
You could use the symmetric constructor here.

```suggestion
                          contentPadding: const EdgeInsets.symmetric(horizontal: 0.0, vertical: 12.0),
```